### ISSUE TITLE
onPostStart extension not called when server is initialized

### ISSUE THREAD

**Issue by jonathansamines** [hapijs/hapi#4308]
#### Support plan

* *is this issue currently blocking your project?* (yes/no): No
* *is this issue affecting a production system?* (yes/no): No

#### Context

* *node version*: 14.17.6
* *module version with issue*: 20.2.1
* *last module version without issue*: -
* *environment* (e.g. node, browser, native): node
* *used with* (e.g. hapi application, another framework, standalone, ...): hapi application
* *any other relevant information*: -

#### What are you trying to achieve or the steps to reproduce?

Hapi is not calling the `onPostStart` server extension when the server is initialized but succeeds if started.

```js
'use strict';

const Hapi = require('@hapi/hapi');
const assert = require('assert');

async function run() {
    const tasks = [];
    const serverA = Hapi.server();

    function createTask(name) {
        return function task() {
            tasks.push(name);
        }
    }

    serverA.ext('onPreStart', createTask('onPreStart'));
    serverA.ext('onPostStart', createTask('onPostStart'));
    serverA.ext('onPreStop', createTask('onPreStop'));
    serverA.ext('onPostStop', createTask('onPostStop'));

    await serverA.initialize();
    await serverA.stop();

    assert.deepStrictEqual(tasks, ['onPreStart', 'onPostStart', 'onPreStop', 'onPostStop']);
}

run()
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });
```

#### What was the result you got?

The test fails wen the server is initialized but succeeds if started.

```bash
+ actual - expected

  [
    'onPreStart',
-   'onPostStart',
    'onPreStop',
    'onPostStop'
  ]
    at run (/workspaces/hapi-test-sandbox/server-extension-onpoststart.js:24:12) {
  generatedMessage: true,
  code: 'ERR_ASSERTION',
  actual: [Array],
  expected: [Array],
  operator: 'deepStrictEqual'
}
```

#### What result did you expect?

The test should always succeed. When testing plugins that add a `onPostStart` extension, we are currently forced to use `server.start()` as opposed to `server.initialize()` which is simpler to deal with in tests.

**Comment by devinivy** [comment]
I am closing this since it's been a little while and I think we've worked it out as far as we can for now, having uncovered that this is the intentional behavior.  My suggestion is to a. go ahead and start the server, or b. pass hapi a `listener` that is patched such that it cannot start, or c. use the `'start'` event and manually emit it within your tests.  If you think these are insufficient solutions, feel free to make a specific feature request based on what we've learned, and we can garner some feedback on it 👍
